### PR TITLE
Add bare-metal driver for eMIOS ICU for S32K344

### PR DESCRIPTION
**Pull request by Dat-NguyenDuy**
This add bare-metal driver for eMIOS ICU and specific configuration for S32K344. EMIOS ICU driver will be used
for PWM pulse capture

### CODE REVIEW

**Review by decsny — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Review by mmahadevan108 — CHANGES_REQUESTED** [review]
Will wait for @manuargue approval before merging this PR

**Review by manuargue — COMMENTED** [review]
> Will wait for @manuargue approval before merging this PR

thanks, there are some comments to be addressed that will affect the HAL part. I'd say let's wait for the zephyr PR to be approved first before merging this one in case updates are needed.

File: s32/soc/s32k344/include/Emios_Icu_Ip_SA_Init_PBcfg.h
Comment (manuargue) [inline s32/soc/s32k344/include/Emios_Icu_Ip_SA_Init_PBcfg.h:110]: is it possible to generate the configuration for all emios instances and channels, so that we reduce the number of patches needed over the generated code?

File: s32/drivers/s32k3/Icu/CMakeLists.txt
Comment (manuargue) [inline s32/drivers/s32k3/Icu/CMakeLists.txt:16]: cannot we have a single Kconfig option for this PWM driver and use the `CONFIG_PWM_CAPTURE` to conditionally build capture support?

File: s32/soc/s32k344/include/Emios_Icu_Ip_SA_Init_PBcfg.h
Comment (Dat-NguyenDuy) [reply s32/soc/s32k344/include/Emios_Icu_Ip_SA_Init_PBcfg.h]: I've update with generated code for all instances and channel

File: s32/drivers/s32k3/Icu/CMakeLists.txt
Comment (Dat-NguyenDuy) [reply s32/drivers/s32k3/Icu/CMakeLists.txt]: I've updated

**Review by manuargue — APPROVED** [review]

**Review by mmahadevan108 — APPROVED** [review]

**Pull request merged**
